src/offset.js

```diff
@@ -63,6 +63,12 @@ jQuery.offset = {
 			options.using.call( elem, props );
 
 		} else {
+			if ( typeof props.top === "number" ) {
+				props.top += "px";
```

Comment:
Why not append `"px"` at the point of definition instead? That should be smaller.

Reply:
Does UI expect to have bare numbers here?

Reply:
Yes, unfortunately. We could redefine `.offset()` to take non-px values but that seems like a bigger change with even more consequences. On the `.css()` API it seems more reasonable to insist on strings with units but here the numbers are always supposed to represent pixels. If we require `.offset()` to give strings with `px` appended then I can see people wanting to use arbitrary units.

Reply:
Right, makes sense. I'd experiment whether something like `[ "top", "left" ].forEach(...)` isn't smaller but if it's not, LGTM, especially for jQuery 3.x.

We could try to clean that up for 4.0, just please, before merging this one, create an issue with the "Blocker" label & the `4.0.0` milestone so that we don't forget.

Reply:
Timmy created an issue so I have no further reservations.